### PR TITLE
Updates to Thompson-Eidhammer microphysics and MPAS to support future microphysics features

### PR DESCRIPTION
**Pull request by AndersJensen-NOAA**
- Adds water- and ice-friendly aerosols passive tracers (nwfa and nifa) to the initial conditions and model code
- Adds water- and ice-friendly aerosols passive tracers (nwfa and nifa) to lateral boundary conditions
- Updates lookup tables for Thompson microphysics (**hail-aware microphysics will not work yet**)
- Adds subroutines to read in monthly climatology data for water- and ice-friendly aerosols for initial and lateral boundary conditions. **Note, currently using a crude interpolation**
- Points to the Thompson-Eidhammer microphysics here: https://github.com/AndersJensen-NOAA/GSL_cloud_physics.git using branch develop. This version is mostly updated to be consistent with WRFv4.5.
- **After this merge, the submodule version of Thompson will need to be used, so a command like `git submodule update --init --recursive` will need to be run after cloning MPAS**
- Tables needed for aerosol data are in the submodule folder and need to be manually linked to the **run** directory before creating initial and boundary conditions.

Testing: one 24-h 15-km test run on Jet with GNU compilers with aerosols initialized and advected, but not coupled to the microphysics.

### CODE REVIEW

File: src/core_atmosphere/physics/Makefile
Comment (AndersJensen-NOAA) [inline src/core_atmosphere/physics/Makefile]: Currently the submodule files are copied to physics_wrf before compiling, but this should be improved so that the submodule files are compiled in the submodule directory and linked appropriately.

File: src/core_atmosphere/Registry.xml
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml:1899]: @AndersJensen-NOAA  add "bl_mynn_in" as well in the packages. From what I gather, I think it matters for making sure these arrays are used in the pbl driver, i.e., see the "ni" portions.

**Review by joeolson42 — APPROVED** [review]

**Review by tanyasmirnova — APPROVED** [review]

File: src/core_atmosphere/physics/Makefile
Comment (AndersJensen-NOAA) [reply src/core_atmosphere/physics/Makefile]: @jderrico-noaa This is outdated now. I added a Makefile in my last commit.

**Review by jderrico-noaa — APPROVED** [review]
LGTM! if the packages @joeolson42 mentioned above need to be added we can wait on that but otherwise I'm good to merge

**Comment by AndersJensen-NOAA** [comment]
> LGTM! if the packages @joeolson42 mentioned above need to be added we can wait on that but otherwise I'm good to merge

Let's handle those in @joeolson42 's PR. There version of MYNN I'm testing with doesn't have nc as an input in the driver.

**Comment by jderrico-noaa** [comment]
> > LGTM! if the packages @joeolson42 mentioned above need to be added we can wait on that but otherwise I'm good to merge
> 
> Let's handle those in @joeolson42 's PR. There version of MYNN I'm testing with doesn't have nc as an input in the driver.

Sounds good.  I'm going to enable squash merging for the repo and then I'll merge this one

**Pull request merged**
